core/src/test/java/io/questdb/cutlass/pgwire/PGSecurityTest.java

```diff
@@ -202,6 +206,46 @@ public void testAllowsSelect() throws Exception {
         });
     }
 
+    @Test
+    public void testInitialPropertiesParsedCorrectly() throws Exception {
+        // there was a bug where a value of each property was also used as a key for a property created out of thin air.
+        // so when a client sends a property with a value set to "user" then a buggy pgwire parser would create
+        // also a key "user" out of thin air with a value set as the next key. Example:
+        // 2022-05-17T16:39:18.308689Z I i.q.c.p.PGConnectionContext property [name=user, value=admin] <-- this is a legit property
+        // 2022-05-17T16:39:18.308707Z I i.q.c.p.PGConnectionContext property [name=admin, value=database] <-- this is a property "invented" by a buggy pgwire parser
+        // 2022-05-17T16:39:18.308724Z I i.q.c.p.PGConnectionContext property [name=database, value=qdb] <-- a legit property set by a client
+        // 2022-05-17T16:39:18.308789Z I i.q.c.p.PGConnectionContext property [name=qdb, value=client_encoding] <-- again, a property created out of thin air
+
+        // so this test sets a property to "user" and check authentication still succeed. it would fail on a buggy pgwire parser
+        // because the out of thin air property would overwrite the user set by the client. Example:
+        // 2022-05-17T15:58:38.973955Z I i.q.c.p.PGConnectionContext property [name=user, value=user] <-- client indicates username is "user"
+        // 2022-05-17T15:58:38.974236Z I i.q.c.p.PGConnectionContext property [name=user, value=database] <-- buggy pgwire parser overwrites username with out of thin air value
+        assertMemoryLeak(() -> {
+            try (
+                    final PGWireServer ignored = createPGServer(1);
+                    // Postgres JDBC clients ignores unknown properties and does not send them to a server
+                    // so have to use a property which actually exists
+                    final Connection connection = getConnectionWithCustomProperty(PGProperty.OPTIONS.getName(), "user");
+            ) {
+                // no need to assert anything, if we manage to create a connection then it's already a success!
```

Comment:
what about checking that properties "out of think air" are not created?

Reply:
that would be great! unfortunately, the code throws away all properties except user(name).  it only logs them. see: https://github.com/questdb/questdb/blob/1687fba77b67c886364da3d67c1044b2d744eb3a/core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java#L1978-L1995

and asserting on logging would be even more awkward than the current test.